src/main/java/uk/gov/pay/api/model/directdebit/agreement/CreateMandateRequest.java

```diff
@@ -5,19 +5,29 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import io.swagger.annotations.ApiModel;
 import io.swagger.annotations.ApiModelProperty;
+import org.hibernate.validator.constraints.Length;
+import uk.gov.pay.api.validation.ValidReturnUrl;
+
+import javax.validation.constraints.NotNull;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @ApiModel(value = "CreateMandateRequest", description = "The Payload to create a new Agreement")
 public class CreateMandateRequest {
 
+    @NotNull
+    @ValidReturnUrl
+    @Length(max = 2000, message = "Must be less than or equal to {max} characters length")
     @JsonProperty(value = "return_url")
     private String returnUrl;
-    
+
+    @NotNull
+    @Length(min = 1, max = 255, message = "Must have a size between {min} and {max}")
```

Comment:
I think `@NotBlank` is better as it will prevent blank strings too. Then you'd only need the a max length

Reply:
fixed